tree/30_generic_methods/permissions_posix_acl_entry_parent.cf

```diff
@@ -0,0 +1,101 @@
+                                      "${user}",
+                                      "${group}",
+                                      "${other}",
+                                      "${parent_permissions_user}",
+                                      "${parent_permissions_group}",
+                                      "${parent_permissions_other}"
+                                    };
+      "report_param"        string => join("_", args);
+      "full_class_prefix"   string => canonify("permissions_posix_acl_entry_parent_${report_param}");
+      "class_prefix"        string => string_head("${full_class_prefix}", "1000");
+      "prefixes"            slist  => { "${old_class_prefix}", "${class_prefix}" };
+
+
+      # Sub methods calls data
+      "inner_base_prefix"                  string => "permissions_acl_entry";
+      "parents_args[${parents}]"           slist => {
+                                                       "${parents}",
+                                                       "false",
+                                                       "${parent_permissions_user}",
+                                                       "${parent_permissions_group}",
+                                                       "${parent_permissions_other}"
+                                                     };
+      "parents_param[${parents}]"          string => join("_", "${parents_args[${parents}]}");
+      "parents_class_prefixes[${parents}]" string => canonify("${inner_base_prefix}_${parents_param[${parents}]}");
+
+  classes:
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+    pass3::
+      "error_parents"    expression => "${parents_class_prefixes[${parents}]}_error";
+      "repaired_parents" expression => "${parents_class_prefixes[${parents}]}_repaired";
+
+
+  methods:
+    pass2::
+      "${parents}" usebundle => permissions_acl_entry(
+                                  "${parents}",
+                                  "false", 
+                                  "${parent_permissions_user}",
+                                  "${parent_permissions_group}",
+                                  "${parent_permissions_other}"
+                                );
+
+      "${path}" usebundle => permissions_acl_entry(
+                                  "${path}",
+                                  "${recursive}", 
+                                  "${user}",
+                                  "${group}",
+                                  "${other}"
+                                );
+    pass3::
+      "failure"  usebundle => _classes_failure("${prefixes}"),
+                ifvarclass => "error_parents";
+      "repaired" usebundle => _classes_failure("${prefixes}"),
+                ifvarclass => "repaired_parents";
+
+  commands:
+      "/usr/share/ncf/tree/10_ncf_internals/modules/promises/get_parent.sh"
```

Comment:
this will only work on root server, you need to use `${sys.workdir}/modules/promises/get_parent.sh`